fhir-bulkimportexport/src/main/java/com/ibm/watson/health/fhir/bulkexport/BulkExportBatchLet.java

```diff
@@ -149,250 +160,230 @@
             searchContext.setPageNumber(pageNum);
             FHIRTransactionHelper txn = new FHIRTransactionHelper(fhirPersistence.getTransaction());
             txn.begin();
             persistenceContext = FHIRPersistenceContextFactory.createPersistenceContext(null, searchContext);
             List<Resource> resources = fhirPersistence.search(persistenceContext, resourceType);
             txn.commit();
-              
-            pageNum++;
 
-            int count = 0;
-            
             if (resources != null) {
                 for (Resource res : resources) {
                     if (stopRequested) {
-                        exitStatus = "Export stopped!" + " exported: " + exported;
                         break;
                     }
-                    count++;
-                    exported++;
-                    String ndJsonLine = res.toString().replace("\r", "").replace("\n", "");
-                    if (combinedJsons == null) {
-                        combinedJsons = ndJsonLine;
-                    } else {
-                        if (combinedJsons.length() < 10) {
-                            // if isSingleCosObject, then we don't need "," for the first record.
-                            combinedJsons = combinedJsons + ndJsonLine;
-                        } else {
-                            combinedJsons = combinedJsons + "," + "\r\n" + ndJsonLine;
-                        }
+                    if (res == null) {
+                        continue;
                     }
-
-                    if (count == cosBatchSize) {
-                        if (isSingleCosObject) {
-                            if (pageNum <= searchContext.getLastPageNumber()) {
-                             // There is more contents to come for the single Cos object, so add "," to the end.
-                                combinedJsons = combinedJsons + ",";
-                                // only push to COS if the combinedJsons is greater than the minimal COS part size.
-                                if (combinedJsons.getBytes(StandardCharsets.UTF_8).length >= Constants.COS_PART_MINIMALSIZE) {
-                                    pushFhirJsons2Cos(combinedJsons);
-                                    count = 0;
-                                    combinedJsons = null;
-                                }
-                            } else {
-                             // There is no more contents to come for the single Cos object, so add "]" to the end and push to COS.
-                                combinedJsons = combinedJsons + "\r\n]";
-                                pushFhirJsons2Cos(combinedJsons);
-                                count = 0;
-                                combinedJsons = null;
-                            }   
+                    try {
+                        FHIRGenerator.generator(Format.JSON).generate(res, bufferStream);
+                        bufferStream.write("\r\n".getBytes());
```

Comment:
Please use a constant for `\r\n` so it can more easilly be changed to `\n` in the future ;-)

Reply:
haha, good point! will change